snap/hooks/install

```diff
@@ -1,28 +1,45 @@
-#!/bin/bash
+#!/usr/bin/env bash
+# Installation hook for charmed-mongodb snap
 
 set -eux
 
+export CONF="${SNAP_DATA}"/etc/mongod
+export PBM_CONF="${SNAP_DATA}"/etc/pbm
+export DATA="${SNAP_COMMON}"/var/lib/mongodb
+export LOGS="${SNAP_COMMON}"/var/log/mongodb
+export MONGO_CONFIG_FILE="${CONF}"/mongod.conf
+
 # Create the necessary parent directories 
-mkdir -p "${SNAP_DATA}/etc/pbm/"
-mkdir -p "${SNAP_DATA}/etc/mongod/"
-mkdir -p "${SNAP_COMMON}/var/lib/mongodb/"
-mkdir -p "${SNAP_COMMON}/var/log/mongodb/"
+mkdir -p $DATA
+mkdir -p $LOGS
+mkdir -p $PBM_CONF
+mkdir -p $CONF
+
+# If we just created the directory, we set up permissions
+if [ stat -c '%u' "$DATA" == 0 -o ! -d "$DATA" ]; then
```

Comment:
1. how can `! -d "$DATA"` ever evaluate to `True` given `$DATA` was potentially created right before?
2. we could perhaps defer the creation of `$DATA` and simplify the condition with:
```
if [ ! -d "${f}" ]; then
    mkdir -p "${DATA}"
    chmod -R 770 "${SNAP_COMMON}"
    chmod 750 "${DATA}"
fi
```

Reply:
Indeed I don't think `-d "$DATA" can be false here.
The important check here is `stat -c '%u' "$DATA" == 0` to check the owner.
If `root` is the owner, we should update the permissions, otherwise it means we are installing on existing data and we should not change this.